### PR TITLE
Replumb AssumeRole (AWS)

### PR DESCRIPTION
**Pull request by 2uasimojo**
Simplify the AssumeRole flow: Rather than doing it via `credential_process` as a callback from within the creds file used by the provision pod, flatten this out so the AssumeRole is done implicitly by the AWS SDK.

This flow remains unchanged:

The clusterdeployment controller:
- Copies the service provider secret into the CD namespace
- Creates an AWS credentials secret
- Creates the provision pod

The provision pod:
- Loads the credentials secret
- Projects the AWS config therein onto the file system
- Invokes the installer

The installer:
- Creates an AWS client using that config file
- Proceeds with installation

Before this commit:
The AWS config contained a `credential_process` which invoked `hiveutil install-manager aws-credentials` which...
- Loaded the service provider secret
- Created an AWS client
- Used the client to AssumeRole and generate credentials with a 15m expiration
- Printed the credentials to stdout in the format expected by AWS.

Per AWS docs[1], the SDK will automatically rerun the `credential_process` before the expiration time to refresh the creds.

With this commit:
The clusterdeployment controller loads the service provider secret and folds it into the AWS config as a separate profile, referenced from the default via `source_profile`:

```
[default]
source_profile = source
role_arn = arn:aws:iam::123456789012:role/assume-role-customer

[profile source]
aws_access_key_id: ABCDEFGHIJKLMNOPQRST
aws_secret_access_key: 1234567890abcdefghijklmnopqrstuvwxyz0123
role_arn = arn:aws:iam::210987654321:role/assume-role-provider
```

Per AWS docs[2], the SDK will use the source creds to AssumeRole to generate temporary creds, which it will automatically refresh as they expire -- i.e. natively performing the same function as `hiveutil install-manager aws-credentials`.

As a security measure, we will now check AWS config/credential files for `credential_process`, and explode if found. However, since existing clusters in the field may still be configured with the old mechanism in the relevant Secrets, we will convert such Secrets to use the new mechanism.

[1] https://docs.aws.amazon.com/cli/v1/userguide/cli-configure-sourcing-external.html
[2] https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-role.html

[HIVE-2485](https://issues.redhat.com//browse/HIVE-2485)
[HIVE-2529](https://issues.redhat.com//browse/HIVE-2529)

### CODE REVIEW

**Comment by cebarks** [comment]
This has been assigned CVE-2024-25133 by Red Hat Product Security.

**Comment by 2uasimojo** [comment]
/override ci/prow/security

Addressed via #2308

File: pkg/controller/clusterdeployment/clusterprovisions.go [hunk 1]
```diff
@@ -774,7 +774,7 @@ func (r *ReconcileClusterDeployment) setupAWSCredentialForAssumeRole(cd *hivev1.
 		return nil
 	}
 
-	return install.AWSAssumeRoleCLIConfig(r.Client, cd.Spec.Platform.AWS.CredentialsAssumeRole, install.AWSAssumeRoleSecretName(cd.Name), cd.Namespace, cd, r.scheme)
+	return install.AWSAssumeRoleConfig(r.Client, cd.Spec.Platform.AWS.CredentialsAssumeRole, install.AWSAssumeRoleSecretName(cd.Name), cd.Namespace, cd, r.scheme)
```
[the remark this answers is not in the document]
Comment (2uasimojo) [reply pkg/controller/clusterdeployment/clusterprovisions.go:777]: Deprovision path is below...

File: pkg/controller/clusterdeprovision/clusterdeprovision_controller.go [hunk 2]
```diff
@@ -478,6 +478,6 @@ func (r *ReconcileClusterDeprovision) setupAWSCredentialForAssumeRole(cd *hivev1
 		return nil
 	}
 
-	return install.AWSAssumeRoleCLIConfig(r.Client, cd.Spec.Platform.AWS.CredentialsAssumeRole, install.AWSAssumeRoleSecretName(cd.Name), cd.Namespace, cd, r.scheme)
+	return install.AWSAssumeRoleConfig(r.Client, cd.Spec.Platform.AWS.CredentialsAssumeRole, install.AWSAssumeRoleSecretName(cd.Name), cd.Namespace, cd, r.scheme)
```
Comment (2uasimojo) [inline pkg/controller/clusterdeprovision/clusterdeprovision_controller.go:481]: ...here

**Comment by jharrington22** [comment]
Over LGTM @2uasimojo

**Review by jstuever — APPROVED** [review]
/lgtm

**Comment by 2uasimojo** [comment]
MachinePool scaling, hopefully a flake

/test e2e-gcp

**Comment by 2uasimojo** [comment]
@jharrington22  agrees to merge. 

/override "Red Hat Konflux / hive-on-pull-request"

/hold cancel

**Pull request merged**